sdk/java-sdk-protobuf-testkit/src/main/scala/kalix/javasdk/testkit/impl/EventingTestKitImpl.scala

```diff
@@ -150,37 +150,48 @@ class TopicImpl(private val probe: TestProbe, codec: MessageCodec) extends Topic
     anyFromMessage(msg.getMessage)
   }
 
-  override def expectOneTyped[T <: GeneratedMessageV3](clazz: Class[T]): TestKitMessage[T] =
+  override def expectOneTyped[T](clazz: Class[T]): TestKitMessage[T] =
     expectOneTyped(clazz, DefaultTimeout)
 
-  override def expectOneTyped[T <: GeneratedMessageV3](clazz: Class[T], timeout: time.Duration): TestKitMessage[T] = {
+  override def expectOneTyped[T](clazz: Class[T], timeout: time.Duration): TestKitMessage[T] = {
     val msg = probe.expectMsgType[EmitSingleCommand]
     val metadata = new MetadataImpl(msg.getMessage.getMetadata.entries)
-    val decodedMsg = codec.decodeMessage(ScalaPbAny(typeUrlFor(metadata), msg.getMessage.payload))
+    val scalaPb = ScalaPbAny(typeUrlFor(metadata), msg.getMessage.payload)
+
+    val decodedMsg = if (typeUrlFor(metadata).startsWith(JsonSupport.KALIX_JSON)) {
+      val objectMapper = JsonSupport.getObjectMapper
+      objectMapper.readerFor(clazz).readValue(msg.getMessage.payload.toByteArray)
```

Comment:
We can use `JsonSupport.decodeJson` directly and it has already some extra logic to deal with primitive bytes

Reply:
Yeah, was the first approach actually but somehow it fails when deserializing the message.. as if the format was not correct. Haven't found the reason why. Will give it another try.

Reply:
Have you tried it again? 
Was that related to the extra characters in front of the payload?

Reply:
So, after some experiments, I think it makes sense to not use `JsonSupport.decodeJson` here as that expects the payload to be encoded with the special tag. However, when we publish the messages out to a broker we don't include such tag, and that's why using the `decodeJson` would break here. To make it work, we would need to manually wrap the payload with `AnySupport.encodingBytesPrimitive` in line 271 and then we could use the `decodeJson` but I don't see a point in doing that.